mlflow/langchain/runnables.py

```diff
@@ -0,0 +1,229 @@
+    lc_runnables_types,
+    pickable_runnable_types,
+)
+from mlflow.utils.file_utils import mkdir
+
+_RUNNABLE_SEQUENCE_STEPS_FILE_NAME = "steps.yaml"
+
+try:
+    from langchain.prompts.loading import type_to_loader_dict as prompts_types
+except ImportError:
+    prompts_types = {"prompt", "few_shot_prompt"}
+
+
+def _load_model_from_path(path: str, model_config=None):
+    from langchain.chains.loading import load_chain
+    from langchain.chains.loading import type_to_loader_dict as chains_type_to_loader_dict
+    from langchain.llms.loading import get_type_to_cls_dict as llms_get_type_to_cls_dict
+    from langchain.llms.loading import load_llm
+    from langchain.prompts.loading import load_prompt
+
+    model_load_fn = model_config.get(_MODEL_LOAD_KEY)
+    if model_load_fn == _RUNNABLE_LOAD_KEY:
+        return _load_runnables(path, model_config)
+    if model_load_fn == _BASE_LOAD_KEY:
+        return _load_base_lcs(path, model_config)
+    if path.endswith(".pkl"):
+        return _load_from_pickle(path)
+    # Load runnables from config file
+    if path.endswith(".yaml"):
+        config = _load_from_yaml(path)
+    elif path.endswith(".json"):
+        config = _load_from_json(path)
+    else:
+        raise MlflowException(f"Cannot load runnable without a config file. Got path {path!s}.")
+    _type = config.get("_type")
+    if _type in chains_type_to_loader_dict:
+        return load_chain(path)
+    elif _type in prompts_types:
+        return load_prompt(path)
+    elif _type in llms_get_type_to_cls_dict():
+        return load_llm(path)
+    elif _type in custom_type_to_loader_dict():
+        return custom_type_to_loader_dict()[_type](config)
+    raise MlflowException(f"Unsupported type {_type} for loading.")
+
+
+def _load_runnable_with_steps(file_path: Union[Path, str], model_type: str):
+    """
+    Load the model
+
+    :param file_path: Path to file to load the model from.
+    :param model_type: Type of the model to load.
+    """
+    from langchain.schema.runnable import RunnableParallel, RunnableSequence
+
+    # Convert file to Path object.
+    load_path = Path(file_path) if isinstance(file_path, str) else file_path
+    if not load_path.exists():
+        raise FileNotFoundError(f"File {load_path!s} does not exist.")
+    if not load_path.is_dir():
+        raise ValueError(f"File {load_path!s} must be a directory.")
+
+    steps_conf_file = os.path.join(load_path, _RUNNABLE_SEQUENCE_STEPS_FILE_NAME)
+    if not os.path.exists(steps_conf_file):
+        raise MlflowException(
+            f"File {steps_conf_file} must exist in order to load the RunnableSequence."
+        )
+    steps_conf = _load_from_yaml(steps_conf_file)
+    if model_type == RunnableSequence.__name__:
+        steps = []
+    elif model_type == RunnableParallel.__name__:
+        steps = {}
+    else:
+        raise MlflowException(f"Unsupported model type {model_type}. ")
+    for file in sorted(os.listdir(load_path)):
```

Comment:
if the runnable has more than 10 steps, will this result in weird sorting? e.g`["0", "1", "10", "2", "3" ...]`